physics/eclipse_test.cpp

```diff
@@ -0,0 +1,167 @@
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "physics/solar_system.hpp"
+#include "quantities/quantities.hpp"
+#include "quantities/si.hpp"
+#include "integrators/symplectic_runge_kutta_nyström_integrator.hpp"
+#include "geometry/epoch.hpp"
+#include "geometry/named_quantities.hpp"
+#include "quantities/elementary_functions.hpp"
+#include "geometry/grassmann.hpp"
+#include "testing_utilities/numerics.hpp"
+
+namespace principia {
+
+using astronomy::ICRFJ2000Equator;
+using geometry::JulianDate;
+using integrators::McLachlanAtela1992Order5Optimal;
+using quantities::si::Minute;
+using quantities::si::Metre;
+using quantities::si::Nano;
+using quantities::si::Milli;
+using quantities::si::Kilo;
+using quantities::ArcCos;
+using testing_utilities::AbsoluteError;
+using ::testing::Gt;
+using ::testing::Lt;
+using ::testing::AllOf;
+
+namespace physics {
+
+class EclipseTest : public testing::Test {
+ protected:
+  EclipseTest() {
+    solar_system_1950_.Initialize(
+        SOLUTION_DIR / "astronomy" / "gravity_model.proto.txt",
+        SOLUTION_DIR / "astronomy" /
+            "initial_state_jd_2433282_500000000.proto.txt");
+  }
+
+  SolarSystem<ICRFJ2000Equator> solar_system_1950_;
+};
+
+TEST_F(EclipseTest, Dummy) {
+  auto ephemeris = solar_system_1950_.MakeEphemeris(
+      McLachlanAtela1992Order5Optimal<Position<ICRFJ2000Equator>>(),
+      45 * Minute, 5 * Milli(Metre));
+
+  ephemeris->Prolong(JulianDate(2433374.5));  // Prolong until date of eclipse
+                                              // (Eclipse was 1950-04-02 but JD
+                                              // is 1950-04-03:00:00:00)
+  // pass body to Ephemeris.trajectory
+  auto sun = ephemeris->bodies()[solar_system_1950_.index("Sun")];
+  auto earth = ephemeris->bodies()[solar_system_1950_.index("Earth")];
+  auto moon = ephemeris->bodies()[solar_system_1950_.index("Moon")];
+
+  // Massive_body eventually needs radius information. Or non-hardcoded data
+  // pulled from
+  // https://github.com/mockingbirdnest/Principia/blob/master/astronomy/gravity_model.proto.txt
+  auto r_sun = 696000.0 * Kilo(Metre);
+  auto r_earth = 6378.1363 * Kilo(Metre);
+  auto r_moon = 1738.0 * Kilo(Metre);
+
+  // Dates are TDB Julian Day for 1948-04-02
+  auto P1 = JulianDate(2433374.25788409);  // 18:10:49 UT
+  auto U1 = JulianDate(2433374.29850909);  // 19:09:19
+  auto U2 = JulianDate(2433374.354979);    // 20:30:38
+  auto U3 = JulianDate(2433374.37367113);  // 20:57:33
+  auto U4 = JulianDate(2433374.43016419);  // 22:18:54
+  auto P4 = JulianDate(2433374.47075446);  // 23:17:21
+
+  // Lunar eclipse -- U1 (time is start of umbral phase)
+  auto q_sun = ephemeris->trajectory(sun)->EvaluatePosition(U1, nullptr);
```

Comment:
`auto const` for all the positions.  Also, the `nullptr` at the end is a bit mysterious, so it's best to comment it thus:
``
EvaluatePosition(U1, /*hint=*/nullptr);
``

Same for the following two lines.